js/test/transactions.spec.js

```diff
@@ -0,0 +1,169 @@
+/** ******************************************************************************
+ *  (c) 2020 Zondax GmbH
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ ******************************************************************************* */
+
+import FlowApp from "../src";
+import TransportNodeHid from "@ledgerhq/hw-transport-node-hid";
+import chai, { expect } from "chai"
+import jsSHA from "jssha";
+import {ec as EC} from "elliptic";
+import fs from "fs";
+
+jest.setTimeout(1000000);
+
+const CHAIN_ID_PAGE_COUNT = 1;
+const REF_BLOCK_PAGE_COUNT = 2;
+const GAS_LIMIT_PAGE_COUNT = 1;
+const PROP_KEY_ADDRESS_PAGE_COUNT = 1;
+const PROP_KEY_ID_PAGE_COUNT = 1;
+const PROP_KEY_SEQNUM_PAGE_COUNT = 1;
+const PAYER_PAGE_COUNT = 1;
+const AUTHORIZER_PAGE_COUNT = 1;
+const ACCEPT_PAGE_COUNT = 1;
+
+const PAGE_SIZE = 34;
+
+function getArgumentPageCount(arg) {
+    if (arg.type == "Array") {
+        return getArgumentsPageCount(arg.value);
+    }
+
+    if (arg.type == "String") {
+        const count = Math.ceil(arg.value.length / PAGE_SIZE);
+        return count;
+    }
+
+    if (arg.type == "Optional" && arg.value != null) {
+        return getArgumentPageCount(arg.value);
+    }
+
+    return 1;
+}
+
+function getArgumentsPageCount(args) {
+    return args.reduce((count, arg) => count + getArgumentPageCount(arg), 0);
+}
+
+function getTransactionBasePageCount(tx) {
+    return (
+        CHAIN_ID_PAGE_COUNT +
+        REF_BLOCK_PAGE_COUNT +
+        GAS_LIMIT_PAGE_COUNT +
+        PROP_KEY_ADDRESS_PAGE_COUNT +
+        PROP_KEY_ID_PAGE_COUNT +
+        PROP_KEY_SEQNUM_PAGE_COUNT +
+        PAYER_PAGE_COUNT +
+        (AUTHORIZER_PAGE_COUNT * tx.authorizers.length) +
+        ACCEPT_PAGE_COUNT
+    );
+}
+
+function getTransactionPageCount(tx) {
+    return getTransactionBasePageCount(tx) + getArgumentsPageCount(tx.arguments);
```

Comment:
this seems rather weird - I would incorporate getArgumentsPageCount into previous function as the ordering is interleaved - arguments are accepted in the middle